node-graph/gcore/src/vector/vector_attributes.rs

```diff
@@ -985,6 +1024,134 @@ impl Vector {
+	/// Negative area indicates clockwise winding, positive indicates counterclockwise.
+	fn compute_signed_area(&self, face: &FoundSubpath) -> f64 {
+		let mut area = 0.0;
+		for edge in &face.edges {
+			let start_pos = self.point_domain.positions()[edge.start];
+			let end_pos = self.point_domain.positions()[edge.end];
+			area += (end_pos.x - start_pos.x) * (end_pos.y + start_pos.y);
+		}
+		area * 0.5
+	}
+
+	/// Find all minimal closed regions (faces) in a branching mesh vector.
+	pub fn find_closed_regions(&self) -> Vec<FoundSubpath> {
+		let mut regions = Vec::new();
+		let mut used_half_edges = HashSet::new();
+
+		// Build adjacency list sorted by angle for proper face traversal
+		let mut adjacency: HashMap<usize, Vec<(SegmentId, usize, bool)>> = HashMap::new();
+
+		for (segment_id, start, end, _) in self.segment_domain.iter() {
+			adjacency.entry(start).or_default().push((segment_id, end, false));
+			adjacency.entry(end).or_default().push((segment_id, start, true));
+		}
+
+		// Sort neighbors by angle to enable finding the "rightmost" path
+		for (&point_idx, neighbors) in adjacency.iter_mut() {
+			let start = self.point_domain.positions()[point_idx];
+
+			neighbors.sort_by(|a, b| {
+				let angles: [f64; 2] = [a, b].map(|&(segment_id, end_idx, reversed)| {
+					let end = self.point_domain.positions()[end_idx];
+					let curve = self.segment_domain.handles()[self.segment_domain.id_to_index(segment_id).unwrap()];
+					let curve = if reversed { curve.reversed() } else { curve };
+					let direction = curve.derivative_at_start(start, end);
+					direction.y.atan2(direction.x)
+				});
+				angles[0].partial_cmp(&angles[1]).unwrap_or(std::cmp::Ordering::Equal)
+			});
+		}
+
+		for (segment_id, start, end, _) in self.segment_domain.iter() {
+			for &reversed in &[false, true] {
+				let (from, to) = if reversed { (end, start) } else { (start, end) };
+				let half_edge_key = (segment_id, reversed);
+
+				if used_half_edges.contains(&half_edge_key) {
+					continue;
+				}
+
+				if let Some(face) = self.find_minimal_face_from_edge(segment_id, from, to, reversed, &adjacency, &mut used_half_edges) {
+					regions.push(face);
+				}
+			}
+		}
+
+		// Filter out outer (counterclockwise) faces, keeping only inner (clockwise) faces
+		regions.retain(|face| self.compute_signed_area(face) <= 0.0);
+		regions
+	}
+
+	/// Helper to find a minimal face (smallest cycle) starting from a half-edge
+	fn find_minimal_face_from_edge(
+		&self,
+		start_segment: SegmentId,
+		from_point: usize,
+		to_point: usize,
+		start_reversed: bool,
+		adjacency: &HashMap<usize, Vec<(SegmentId, usize, bool)>>,
+		used_half_edges: &mut HashSet<(SegmentId, bool)>,
+	) -> Option<FoundSubpath> {
+		let mut path = vec![HalfEdge::new(start_segment, from_point, to_point, start_reversed)];
+		let mut current = to_point;
+		let target = from_point;
+		let mut prev_segment = start_segment;
+
+		let max_iterations = adjacency.len() * 2;
```

Comment:
Why would this be the max iterations? Surely you should only loop `adjacency.len()` times (since you guarantee to not use the same segment multiple times).